server/api.js

```diff
@@ -13,27 +16,34 @@ router.get("/", (_, res) => {
 
 router.get("/clientId", (req, res) => {
 	try {
-		const clientId = process.env.REACT_APP_CLIENT_ID;
+		const clientId = process.env.CLIENT_ID;
 		if (!clientId) {
 			throw new Error("Client ID not found.");
 		}
 		res.json({ clientId }).status(200);
 	} catch (error) {
 		logger.error("Error fetching clientId:", error.message);
-		res.status(500).json({ error: "Internal server error Heni" });
+		res.status(500).json({ error });
 	}
 });
 
+
 router.post("/validation", async (req, res) => {
-	const { token } = req.body;
+	const { token,role } = req.body;
 	try {
 		const client = new OAuth2Client();
 		const ticket = await client.verifyIdToken({
 			idToken: token,
-			audience: process.env.REACT_APP_CLIENT_ID,
+			audience: process.env.CLIENT_ID,
 		});
 		const payload = ticket.getPayload();
-		res.status(200).json({ message: "success" });
+		const { name, email } = payload;
+		let user = await Users.findOne({ where: { email } });
+		if (!user) {
+			persistNewUser(name, email, role, token);
+		} else {
+			res.status(200).json({ message: "success" });
+		}
```

Comment:
Created a bug for what we are doing here:  https://github.com/users/karleenmsrichards/projects/4/views/1?pane=issue&itemId=35131052

Reply:
We always want to return a reponse code

```javascript
let user = await Users.findOne({ where: { email } });
     try {
		if (!user) {
			persistNewUser(name, email, role, token);
		} 
			res.status(200).json({ message: "success" });
		}
} catch(e){
             res.status(500).json({ message: error });
}

```

Reply:
I will add return with my next PR.